homeassistant/components/actiontec/device_tracker.py

```diff
@@ -33,43 +31,40 @@
 )
 
 
-def get_scanner(hass, config):
+def get_scanner(
+    hass: HomeAssistant, config: ConfigType
+) -> ActiontecDeviceScanner | None:
     """Validate the configuration and return an Actiontec scanner."""
     scanner = ActiontecDeviceScanner(config[DOMAIN])
     return scanner if scanner.success_init else None
 
 
-Device = namedtuple("Device", ["mac", "ip", "last_update"])
-
-
 class ActiontecDeviceScanner(DeviceScanner):
     """This class queries an actiontec router for connected devices."""
 
-    def __init__(self, config):
+    def __init__(self, config: ConfigType) -> None:
         """Initialize the scanner."""
-        self.host = config[CONF_HOST]
-        self.username = config[CONF_USERNAME]
-        self.password = config[CONF_PASSWORD]
-        self.last_results = []
+        self.host: str = config[CONF_HOST]
+        self.username: str = config[CONF_USERNAME]
+        self.password: str = config[CONF_PASSWORD]
+        self.last_results: list[Device] = []
         data = self.get_actiontec_data()
         self.success_init = data is not None
         _LOGGER.info("Scanner initialized")
 
-    def scan_devices(self):
+    def scan_devices(self) -> list[str]:
         """Scan for new devices and return a list with found device IDs."""
         self._update_info()
-        return [client.mac for client in self.last_results]
+        return [client.mac_address for client in self.last_results]
 
-    def get_device_name(self, device):
+    def get_device_name(self, device: str) -> str:
```

Comment:
My comment disappeared 😕
```suggestion
    def get_device_name(self, device: str) -> str | None:  # type: ignore[error-code]
```

`device_tracker` is typed incorrectly. `DeviceTracker.async_see` expects `None` and many components return it, it's not an error:
https://github.com/home-assistant/core/blob/8bc75e91a02f9b0a743c26bed1ee31a1d4c96551/homeassistant/components/device_tracker/legacy.py#L462

Reply:
This causes again an "incompatible with supertype" error:
`error: Return type "Optional[str]" of "get_device_name" incompatible with return type "str" in supertype "DeviceScanner"  [override]` ... so not sure if ignoring this is ok :thinking:

Reply:
It's ok, super type should be fixed instead.